plugins/services/src/js/reducers/serviceForm/common/Constraints.js

```diff
@@ -40,13 +45,44 @@ function processTransaction(state, { type, path, value }) {
 
 module.exports = {
   JSONParser(constraints) {
-    // Ignore non-array constraints
     if (!Array.isArray(constraints)) {
-      return [];
+      return [new Transaction(["constraints"], "not-list", ERROR)];
     }
 
     return constraints.reduce(function(memo, item, index) {
-      if (typeof item !== "object") {
+      if (Array.isArray(item)) {
+        memo.push(
+          new Transaction(
+            ["constraints", index, "value"],
+            "value-not-converted-to-object",
+            ERROR
+          )
+        );
+
+        return memo;
+      }
+
+      if (typeof item != "object") {
```

Comment:
🎨  Please always use `!==` or `===` except for the cases in which you really want to convert types.

Reply:
I tend to follow what "You don't know javascript" for this coercion style issues, on his book he says:

https://github.com/getify/You-Dont-Know-JS/blob/master/types%20%26%20grammar/ch4.md

More precisely.
```
Tip: Another place where coercion is guaranteed not to bite you is with the typeof operator. typeof is always going to return you one of seven strings (see Chapter 1), and none of them are the empty "" string. As such, there's no case where checking the type of some value is going to run afoul of implicit coercion. typeof x == "function" is 100% as safe and reliable as typeof x === "function". Literally, the spec says the algorithm will be identical in this situation. So, don't just blindly use === everywhere simply because that's what your code tools tell you to do, or (worst of all) because you've been told in some book to not think about it. You own the quality of your code.
```

I also went on because the linter did not complain. So not really sure this is an issue.

Reply:
Yeah, it's fine as is and you don't need to change it, but I'd prefer us to always use strict checks as these are the things that usually get lost during a refactor. P.S. Comments I flag with `🎨` are purely stylistic and it's up to you wether you want to adjust your implementation.

Reply:
Alright. I will keep 🎨  in mind. I think that by justifying myself I am sharing my 🎨  opinion.